### DISCUSSION TITLE
How to auto-login with Amazon next.js app router?

### DISCUSSION THREAD

**Discussion by drattansingh** [nextauthjs/next-auth#11291] [Help]
I selected the "Login with Amazon" option in [amazon-console](https://developer.amazon.com/dashboard), created my oauth2 security profile and got my clientID and secret.

Then in my /api/auth/[...nextauth]/route.js file:

```
import AmazonOAuth2Provider from "next-auth/providers/auth0"
....
providers: [
AmazonOAuth2Provider({
        clientId: process.env.NEXT_PUBLIC_AMAZON_CLIENT_ID,
        clientSecret: process.env.NEXT_PUBLIC_AMAZON_CLIENT_SECRET,
        scope: 'profile',
        authorizationUrl: 'https://www.amazon.com/ap/oa',
        tokenUrl: 'https://api.amazon.com/auth/o2/token',
        profileUrl: 'https://api.amazon.com/user/profile',
      }),
```

Then I'm calling in my client as follows:

```
import { signIn } from "next-auth/react";
.....
<button onClick={()=>signIn('amazon')} >Log In with Amazon</button>
```

However while I'm not getting any error, it's displaying a login form with option `sign in with autho` 

![ssdsdsdsds](https://github.com/nextauthjs/next-auth/assets/96189286/a1af39b6-e139-46d6-803d-4da471943c2a)

It's not performing the auto login. Any idea on how to solve this problem and get my auto-login working with amazon?

**Comment by RaviAsnani** [comment]
@drattansingh were you able to figure this out?

**Reply by drattansingh** [reply]
yes
